src/Razor/src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -30,6 +31,7 @@ public GenerateCommand(Application parent)
             ExtensionNames = Option("-n", "extension name", CommandOptionType.MultipleValue);
             ExtensionFilePaths = Option("-e", "extension file path", CommandOptionType.MultipleValue);
             RootNamespace = Option("--root-namespace", "root namespace for generated code", CommandOptionType.SingleValue);
+            CSharpLanguageVersion = Option("--csharp-language-version", "csharp language version generated code", CommandOptionType.SingleValue);
```

Comment:
**You:** 

> Also, Roslyn suggested that we avoid dealing with LangVersion directly because there are several factors that impact its "effective" value on a project when run in tooling


**Also You:** 

> CSharpLanguageVersion = Option("--csharp-language-version", "csharp language version generated code", CommandOptionType.SingleValue);

If it's really important that we don't use the language version, we're going to have to figure something out 😆

Reply:
Ya you definitely pin pointed one of my big pain points with this design. In Jason's words `The thing you probably don't also want to be doing is interpreting the LangVersion property directly, since there's all sorts of fun there...`. The one thing that made me feel slightly better is that we've been doing this approach (using their language version facts service) in MVC so it didn't feel 100% awful.

Reply:
OK, it would be cool and good to know what the alternatives are....

Reply:
Talked offline. Going with this approach for now.